cardinal/ecs/abi.go

```diff
@@ -0,0 +1,81 @@
+package ecs
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"reflect"
+	"strings"
+)
+
+var goTypeToSolType = map[string]string{
+	"bool":           "bool",
+	"string":         "string",
+	"uint":           "uint64",
+	"int":            "int64",
+	"common.Address": "address",
+	"uint8":          "uint8",
+	"uint16":         "uint16",
+	"uint32":         "uint32",
+	"uint64":         "uint64",
+}
+
+func GenerateABIType(goStruct any) (*abi.Type, error) {
```

Comment:
This seems like a good candidate for a table driven test.

Reply:
i added more test coverage, but seemed more straightforward to just have a massive struct with all supported types to make sure we can pack & unpack, and then individual tests for edge cases. 

if you still think its valuable to have these be table driven, lmk, i can look into this more